### DISCUSSION TITLE
Public directory accessibility

### DISCUSSION THREAD

**Discussion by braitsch** [nuxt/nuxt#16242] [Questions]
If I put a file (e.g. image.jpg) in /public and build my app, the image will be available at mydomain.com/image.jpg. However if I add a file to /public **after** the project has been built, for example if I upload it via FTP or write to /public via another process running on my VPS, the file is not publicly available. Is there a way to disable this behavior? IIRC files in the /static directory in Nuxt2 were always publicly available regardless of how they got there.

**Comment by procrates** [comment]
Good question! I'd like to know that too. And at which point do they get referenced in the build?

**Comment by TheAlexLichter** [comment]
The downside of the approach used in Nuxt 2 is that you cannot "deploy" you app without the "static folder". Nitro will [copy the public assets over](https://github.com/unjs/nitro/blob/047357f6fbbe6eb8b51288734b62c2f2dd6a460d/src/build.ts#L31-L41), so you'd have to use the `.output` (or `.dist` folder, depending on the preset) instead.

**Comment by braitsch** [comment]
All content in /public should be publicly available regardless of how the assets got in there. I have a form that allows users to upload images. If I save them to /public they're not publicly available. That's ridiculous.

**Reply by TheAlexLichter** [reply]
As I just explained, the `public` folder is only used up until "building" the application. Everything else is in `.output` or `dist` usually.

**Reply by procrates** [reply]
Is this a static generated website? Maybe nuxt generates the links on generation. But I do belive there is a workaround with the server you are running. If it is a cdn then i think it's not possible.

**Reply by braitsch** [reply]
@manniL Files within .output/public are not publicly available if they're added there after the app is built.

**Reply by joni1802** [reply]
I can confirm that files which are added to .output/public after the build process are not accessible. I am really surprised that there is no solution for that. This makes my app currently useless :-(

**Comment by joni1802** [comment]
If anyone is interested I came up with the following solution:
See https://github.com/nuxt/framework/issues/9313#issuecomment-1329686518
